### PR TITLE
refactor(sunjx): refactor advantage calculation logic

### PR DESCRIPTION
**Pull request by Jiaxuan-Sun**
# New advantage_calculator.py Module

## Overview
A new module implementing various advantage calculation methods for reinforcement learning algorithms.

## Abstract Base Class
- **AdvantageCalculator** - Defines a unified interface for all advantage calculators

## Implementations
- **GAECalculator** - Generalized Advantage Estimation (GAE)
- **CPGDCalculator** - Clipped Policy Gradient Optimization with Policy Drift (CPGD)
- **REINFORCECalculator** - REINFORCE algorithm
- **RLOOCalculator** - REINFORCE Leave-One-Out (RLOO)
- **REINFORCEBaselineCalculator** - REINFORCE with baseline
- **GroupNormCalculator** - Group Normalization (supports "grpo" alias)

## Factory Function
- **get_advantage_calculator** - Factory function for instance creation

### CODE REVIEW

File: lightrft/trainer/fast_exp_maker.py [hunk 1]
```diff
@@ -1486,48 +1443,33 @@ def _compute_advantages_and_returns(
...
-                    self.get_advantages_and_returns(
-                        experience.values,
-                        final_reward,
-                        experience.action_mask,
-                        generate_kwargs["gamma"],
-                        generate_kwargs["lambd"],
-                    )
-                )
+            # Prepare helper functions for calculator
+            get_gae_fn = (
```
Comment (puyuan1996) [inline lightrft/trainer/fast_exp_maker.py]: get_cumulative_returns and get_advantages_and_returns should also be moved into their respective AdvantageCalculator classes.

File: lightrft/trainer/advantage_calculator.py [hunk 2]
```diff
@@ -0,0 +1,751 @@
...
+import warnings
+
+from .utils import RunningMoments, compute_clip_fraction
+
+
+# ============================================================================
+# CPGD Utility Functions
+# ============================================================================
+
+def _get_cpgd_advantages_returns(
```
Comment (puyuan1996) [inline lightrft/trainer/advantage_calculator.py]: _get_cpgd_advantages_returns should also be moved into their respective AdvantageCalculator classes.

**Comment by PaParaZz1** [comment]
refactor typo in the PR name

File: lightrft/trainer/fast_exp_maker.py [hunk 3]
```diff
@@ -901,6 +902,11 @@ def __init__(self, *args, packing_samples: bool = False, processor=None, **kwarg
         else:
             self.reward_running_moments = None
 
+        # Initialize advantage calculator
+        advantage_estimator = self.strategy.config.advantage_estimator
+        self.advantage_calculator = get_advantage_calculator(advantage_estimator, strategy=self.strategy)
+        self.advantage_calculator.reward_running_moments = self.reward_running_moments
```
Comment (PaParaZz1) [inline lightrft/trainer/fast_exp_maker.py]: why do we set this variable here, it is not used in any advantage calculator

File: lightrft/trainer/fast_exp_maker.py [hunk 4]
```diff
@@ -1533,48 +1486,27 @@ def _compute_advantages_and_returns(
...
-            elif self.advantage_estimator in ["reinforce", "rloo", "reinforce_baseline", "group_norm"]:
-                # Compute cumulative returns
-                experience.returns = self.get_cumulative_returns(
-                    final_reward, experience.action_mask, generate_kwargs["gamma"]
+            # Compute advantages and returns using calculator
+            experience.advantages, experience.returns, info_dict = (
+                self.advantage_calculator.compute(
+                    experience,
+                    final_reward,
+                    generate_kwargs,
```
Comment (PaParaZz1) [inline lightrft/trainer/fast_exp_maker.py]: We can extract the commonly used gamma parameter from here and define it directly as gamma=xxx.

**Comment by puyuan1996** [comment]
> * RLOO

RLOO: REINFORCE Leave-One-Out
CPGD: Clipped Policy Gradient Optimization with Policy Drift

File: lightrft/trainer/advantage_calculator.py [hunk 5]
```diff
@@ -0,0 +1,722 @@
+"""
+Advantage Calculator Module
+
+This module provides a unified interface for computing advantages and returns
+in reinforcement learning from human feedback (RLHF) and reinforcement
+learning with verifiable rewards (RLVR) workflows. It abstracts
+different advantage estimation methods (GAE, CPGD, REINFORCE, etc.) into a
```
Comment (puyuan1996) [inline lightrft/trainer/advantage_calculator.py]: order follows the previously mentioned

File: lightrft/trainer/advantage_calculator.py [hunk 6]
```diff
@@ -0,0 +1,722 @@
...
+            info_dict["advantage_clip_frac"] = compute_clip_fraction(advantages, clip_val, -clip_val)
+            advantages = torch.clamp(advantages, -clip_val, clip_val)
+
+        return advantages, returns, info_dict
+
+
+class REINFORCECalculator(BaseREINFORCECalculator):
+    """
+    Standard REINFORCE calculator.
+
```
Comment (puyuan1996) [inline lightrft/trainer/advantage_calculator.py]: why we have both  REINFORCECalculator and BaseREINFORCECalculator? should we delete REINFORCECalculator?

File: lightrft/trainer/advantage_calculator.py [hunk 7]
```diff
@@ -0,0 +1,722 @@
...
+    Reference: REINFORCE: Williams, R. J. (1992). Simple statistical gradient-following
+    algorithms for connectionist reinforcement learning. Machine learning, 8(3-4), 229-256.
+    """
+    pass
+
+
+class RLOOCalculator(BaseREINFORCECalculator):
+    """
+    Reward Leave-One-Out (RLOO) calculator.
```
Comment (puyuan1996) [inline lightrft/trainer/advantage_calculator.py]: fix name

File: lightrft/trainer/experience_maker.py [hunk 8]
```diff
@@ -686,3 +690,50 @@ def get_cumulative_returns(
             returns[:, t] = cumulative_return
 
         return returns
+
+    @torch.no_grad()
+    def normalize_advantages_cross_batch(self, experiences: List[Experience]) -> List[Experience]:
```
Comment (puyuan1996) [inline lightrft/trainer/experience_maker.py]: should we move normalize_advantages_cross_batch method  into lightrft/trainer/advantage_calculator.py?
